src/backends/mod.rs

```diff
@@ -35,10 +35,205 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
+use errors::HWTracerError;
+use Tracer;
+use backends::dummy::DummyTracer;
+
 #[cfg(perf_pt)]
 pub mod perf_pt;
 #[cfg(perf_pt)]
 pub use self::perf_pt::PerfPTTracer;
 
+#[derive(Debug, PartialEq)]
+pub enum BackendKind {
+    Dummy,
+    PerfPT,
+}
+
+/// Generic configuration interface for all backends.
+/// A `None` field indicates a default value and attributes which don't apply to a given backend
```

Comment:
"If an attribute is set to `None`, a default value will be filled in later by <whatever component does that>."